pyccl/tests/test_pk2d.py

```diff
@@ -260,3 +260,99 @@ def test_pk2d_parsing():
     with pytest.raises(ValueError):
         ccl.angular_cl(cosmo, lens1, lens1, ells,
                        p_of_k_a=3)
+
+
+def test_pk2d_get_spline_arrays():
+    empty_pk2d = ccl.Pk2D(empty=True)
+
+    # Pk2D needs splines defined to get splines out
+    with pytest.raises(ValueError):
+        empty_pk2d.get_spline_arrays()
+
+
+def test_pk2d_add():
+    x = np.linspace(0.1, 1, 10)
+    log_y = np.linspace(-3, 1, 20)
+    zarr_a = np.outer(x, np.exp(log_y))
+    zarr_b = np.outer(-1*x, 4*np.exp(log_y))
+
+    empty_pk2d = ccl.Pk2D(empty=True)
+    pk2d_a = ccl.Pk2D(a_arr=x, lk_arr=log_y, pk_arr=np.log(zarr_a),
+                      is_logp=True)
+    pk2d_b = ccl.Pk2D(a_arr=2*x, lk_arr=log_y, pk_arr=zarr_b,
+                      is_logp=False)
+    pk2d_b2 = ccl.Pk2D(a_arr=x, lk_arr=log_y+0.5, pk_arr=zarr_b,
+                       is_logp=False)
+
+    # This raises an error because the a ranges don't match
+    with pytest.raises(ValueError):
+        pk2d_a + pk2d_b
+    # This raises an error because the k ranges don't match
+    with pytest.raises(ValueError):
+        pk2d_a + pk2d_b2
+    # This raises an error because addition with an empty Pk2D should not work
+    with pytest.raises(ValueError):
+        pk2d_a + empty_pk2d
+
+    pk2d_c = ccl.Pk2D(a_arr=x, lk_arr=log_y, pk_arr=zarr_b,
+                      is_logp=False)
+
+    # This raises an error because addition is only defined for Pk2D + Pk2D
+    with pytest.raises(ValueError):
+        pk2d_a + 1
+    with pytest.raises(ValueError):
+        1 + pk2d_a
```

Comment:
As above. Maybe this could be a valid call.
Otherwise, this should be a `TypeError`.